src/App.js

```diff
@@ -1,5 +1,79 @@
-function App() {
-  return <div>Hello World!</div>;
+import { useState } from 'react';
+import styled from 'styled-components/macro';
+import GameForm from './components/GameForm';
+import Navigation from './components/Navigation';
+import GamePage from './pages/GamePage';
+import HistoryPage from './pages/HistoryPage';
+import { nanoid } from 'nanoid';
+
+export default function App() {
+  const [players, setPlayers] = useState([]);
+  const [nameOfGame, setNameOfGame] = useState('');
+  const [currentPage, setCurrentPage] = useState('play');
+  const [history, setHistory] = useState([]);
+
+  return (
+    <AppLayout>
+      <h1>Scorekeeper</h1>
+      {currentPage === 'play' && <GameForm onCreateGame={createGame} />}
+
+      {currentPage === 'game' && (
+        <GamePage
+          nameOfGame={nameOfGame}
+          players={players}
+          onResetScores={resetScores}
+          onEndGame={endGame}
+          onMinus={handleMinus}
+          onPlus={handlePlus}
+        />
+      )}
+
+      {currentPage === 'history' && <HistoryPage history={history} />}
+
+      {(currentPage === 'play' || currentPage === 'history') && (
+        <Navigation currentPage={currentPage} onNavigate={setCurrentPage} />
+      )}
+    </AppLayout>
+  );
+
+  function createGame({ nameOfGame, playerNames }) {
+    setNameOfGame(nameOfGame);
+    setPlayers(playerNames.map(name => ({ name, score: 0 })));
+    setCurrentPage('game');
+  }
+
+  function endGame() {
+    setHistory([{ players, nameOfGame, id: nanoid() }, ...history]);
+    setPlayers([]);
+    setNameOfGame('');
+    setCurrentPage('play');
+  }
+
+  function resetScores() {
+    setPlayers(players.map(player => ({ ...player, score: 0 })));
+  }
+
+  function handlePlus(index) {
+    const currentPlayer = players[index];
+    setPlayers([
+      ...players.slice(0, index),
+      { ...currentPlayer, score: currentPlayer.score + 1 },
+      ...players.slice(index + 1),
+    ]);
```

Comment:
```suggestion
    setPlayers(players.map((player,playerIndex) => index === playerIndex { ...player, score: player.score + 1 } : player));
```
Imho this is much nicer than the spreading pattern

Reply:
Yes and no, it's a bit shorter but if not used to it is also a bit hard to read, I think. Not sure though what the course will be used to so I would check with Jerry again which syntax should be used.

Reply:
sure! just my preference

Reply:
I know :) and it's great to get different preferences and views on topics! :)

Reply:
I like both versions. In the new official docs for React, they show `Immer` to keep the updates simple: https://beta.reactjs.org/learn/updating-arrays-in-state#write-concise-update-logic-with-immer. 

That's why we also show it now. With `use-immer`, it is super simple to update such complex states.

Reply:
I think I would like to leave it as it is - for now. 😬 But thank you for your input - both of you!